### ISSUE TITLE
repair_smiles_of_salts can fail

### ISSUE THREAD

**Issue by florian-huber** [matchms/matchms#577]
I ran the following cleaning pipeline on the `GNPS_LIBRARY.mgf` (from https://gnps-external.ucsd.edu/gnpslibrary).

```python
results_folder = "./results_repairing_no_removing"
os.makedirs(results_folder, exist_ok=True)
compound_class_csv_file = "./compound_name_annotation.csv"

workflow = create_workflow(yaml_file_name=os.path.join(results_folder, "library_cleaning.yaml"),
                           query_filters=DEFAULT_FILTERS + REPAIR_ANNOTATION)

pipeline = Pipeline(workflow,
                    logging_file=os.path.join(results_folder, "library_cleaning_log.log"),
                    logging_level="WARNING")

pipeline.run(data_file,
             cleaned_query_file=os.path.join(results_folder, "cleaned_spectra.mgf"))
```

The pipeline crashed with a TypeError (at `3023it`):
```
...\matchms\matchms\filtering\metadata_processing\repair_smiles_of_salts.py:34, in repair_smiles_of_salts(spectrum_in, mass_tolerance)
     32 for ion, not_used_ions in possible_ion_combinations:
     33     ion_mass = get_monoisotopic_neutral_mass(ion)
---> 34     mass_diff = abs(parent_mass - ion_mass)
     35     # Check for Repair parent mass is mol wt did only return 1 spectrum. So not added as option for simplicity.
     36     if mass_diff < mass_tolerance:

TypeError: unsupported operand type(s) for -: 'float' and 'NoneType'
```

**Comment by florian-huber** [comment]
`repair_smiles_of_salts` fails with the following spectrum:
```
{'charge': 1,
 'ionmode': 'positive',
 'smiles': 'OCC1NC(C2NC(C3NC(C(NC(=O)CNC(C(NC1=O)CO)=O)CCCN1O[Fe]456([O]=C1\\C=C(\\CC(O)=O)C)[O]=C(N(O4)CCC2)\\C=C(\\CC(O)=O)C)=O)=O)=O.C(=[O]5)(N(O6)CCC3)\\C=C(\\CC(O)=O)C',
 'pubmed': '27956353',
 'scans': '1',
 'ms_level': '2',
 'instrument_type': 'LC-ESI-Hybrid FT',
 'file_name': 'f.novakj/siderophores/Ferri_ferrichrome_A.mzXML;',
 'peptide_sequence': '*..*',
 'organism_name': 'GNPS-LIBRARY',
 'compound_name': 'Ferrichrome A [M+Fe-2H] M+H',
 'principal_investigator': 'Havlicek',
 'submit_user': 'novakj',
 'confidence': '3',
 'spectrum_id': 'CCMSLIB00001059075',
 'precursor_mz': 1053.32}
 ```

@niekdejonge can you check?
Otherwise let me know and I will dig into it.

**Comment by florian-huber** [comment]
Probably enough to catch the case that the `ion_mass` is `None`?

**Comment by niekdejonge** [comment]
Ah, did not have this before, but had something similar with parent_mass being None as well. Adding an if statement that checks if the ion_mass is not None, before running the code below should fix this.

**Comment by niekdejonge** [comment]
The last ion of the smile in your example is indeed an invalid smiles: C(=[O]5)(N(O6)CCC3)\\C=C(\\CC(O)=O)C
SyntaxError: The parser encountered a bond repetition.

**Comment by niekdejonge** [comment]
Should be fixed with #578 can you merge it if you agree @florian-huber

**Comment by florian-huber** [comment]
Thanks @niekdejonge . I slightly modified the code and merged it.
